### PR TITLE
[feat] Add startPaused setting to consumer

### PR DESCRIPTION
**Pull request by massakam**
### Motivation

The Java client consumer has a setting called `startPaused`. If this is set to true, the created consumer will not fetch messages from the broker until resume is called. Currently, this setting does not seem to exist in the C++ client consumer, so I will add it.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### Documentation

- [x] `doc-not-needed`

### CODE REVIEW

**Comment by BewareMyPower** [comment]
The 3.5.0 candidate 2 is in vote, do you want to include this PR in this release? Or is it acceptable to delay it to 3.6.0?

**Comment by massakam** [comment]
@BewareMyPower Oh, I missed it. Changed the milestone to 3.6.0.

File: tests/BasicEndToEndTest.cc [hunk 1]
```diff
@@ -968,6 +967,60 @@ TEST(BasicEndToEndTest, testMessageListenerPause) {
...
+    for (int i = 0; i < numOfMessages; i++) {
+        std::string messageContent = "msg-" + std::to_string(i);
+        Message msg = MessageBuilder().setContent(messageContent).build();
+        ASSERT_EQ(ResultOk, producer.send(msg));
+    }
+
+    std::this_thread::sleep_for(std::chrono::microseconds(2 * 1000 * 1000));
+    ASSERT_EQ(globalCount, 0);
+    consumer.resumeMessageListener();
+    std::this_thread::sleep_for(std::chrono::microseconds(2 * 1000 * 1000));
```
Comment (BewareMyPower) [inline tests/BasicEndToEndTest.cc:1012]: Use `waitUntil` to avoid sleeping too long, see examples in tests/ProducerTest.cc

**Review by BewareMyPower — APPROVED** [review]

File: tests/BasicEndToEndTest.cc (hunk 1 above)
Comment (massakam) [reply tests/BasicEndToEndTest.cc]: Replaced the second sleep with `waitUntil`. Replacing the first one would be difficult.

**Review by shibd — APPROVED** [review]

**Pull request merged**